### PR TITLE
Feature/sc 29047/implement submitting a learning object in

### PR DESCRIPTION
**Pull request by johngonz0**
Client Changes for submitting a learning object, and some route/service refactor for old "getFirstSubmission" story.

### CODE REVIEW

**Review by curlyLasagna — APPROVED** [review]
lgtm

**Review by Greek — APPROVED** [review]

**Review by stef359 — APPROVED** [review]

**Review by JosueMolinaMorales — APPROVED** [review]

File: src/app/core/learning-object-module/submissions/submissions.routes.ts [hunk 1]
```diff
@@ -11,16 +12,23 @@ export const SUBMISSION_ROUTES = {
             userId
         )}/learning-objects/${encodeURIComponent(learningObjectId)}/submissions`;
     },
+    CHECK_FIRST_SUBMISSION(params: {
```
Comment (mfranz1) [inline src/app/core/learning-object-module/submissions/submissions.routes.ts:15]: If you don't mind, could you do a bloc comment like on 24-29 that just gives a quick overview of what this route is doing

**Review by mfranz1 — COMMENTED** [review]
Good little refactor

File: src/app/core/learning-object-module/submissions/submissions.routes.ts (hunk 1 above)
Comment (johngonz0) [reply src/app/core/learning-object-module/submissions/submissions.routes.ts:15]: sure thing

**Review by mfranz1 — APPROVED** [review]

**Review by shanioffe — APPROVED** [review]
Spectacular 😎

**Review by pzalep1 — APPROVED** [review]

**Pull request merged**